pygmt/clib/loading.py

```diff
@@ -97,16 +98,37 @@ def clib_full_names(env=None):
     """
     if env is None:
         env = os.environ
+
     libnames = clib_names(os_name=sys.platform)  # e.g. libgmt.so, libgmt.dylib, gmt.dll
-    libpath = env.get("GMT_LIBRARY_PATH", "")  # e.g. $HOME/miniconda/envs/pygmt/lib
 
-    lib_fullnames = [os.path.join(libpath, libname) for libname in libnames]
-    # Search for DLLs in PATH if GMT_LIBRARY_PATH is not defined [Windows only]
-    if not libpath and sys.platform == "win32":
+    # list of libraries paths to search, sort by priority from high to low
+    lib_fullnames = []
+
+    # Search for libraries in GMT_LIBRARY_PATH if defined.
+    libpath = env.get("GMT_LIBRARY_PATH", "")  # e.g. $HOME/miniconda/envs/pygmt/lib
+    if libpath:
+        for libname in libnames:
+            lib_fullnames.append(os.path.join(libpath, libname))
+
+    # Search for the library returned by command "gmt --show-library"
+    try:
+        lib_fullpath = sp.check_output(
+            ["gmt", "--show-library"], encoding="utf-8"
+        ).rstrip("\n")
+        lib_fullnames.append(lib_fullpath)
+    except FileNotFoundError:  # command not found
+        pass
```

Comment:
Is there a way to avoid always calling `gmt --show-library` here? I.e. check `GMT_LIBRARY_PATH` or the `libnames` list first to see if we can load the library before resorting to a subprocess call here. Might involve quite a bit of refactoring though.

Reply:
Yes, I also don't like calling `gmt --show-library` here. I'll see if there is a better way.

Reply:
I changed the `clib_full_names` function into a generator function with `yield` statements (see https://realpython.com/introduction-to-python-generators/#understanding-the-python-yield-statement) so it *should* only hit the `gmt --show library` part if the higher priority stuff didn't get hit. Will see how the CI tests go.